HackerNews/HackerNews/CoreData/DataStoreImpl.m

```diff
@@ -0,0 +1,79 @@
+//
+//  DataStoreImpl.m
+//  HackerNews
+//
+//  Created by SonNguyen on 6/20/18.
+//  Copyright © 2018 SonNguyen. All rights reserved.
+//
+
+#import "DataStoreImpl.h"
+#import "AppDelegate.h"
+#import "StoryObject.h"
+#import "Story.h"
+
+
+@interface DataStoreImpl()
+
+@property (strong, nonatomic) NSManagedObjectContext *managedObjectContext;
+
+@end
+
+
+@implementation DataStoreImpl
```

Comment:
Should this class have singleton? Please consider thread safety when dealing with Coredata. CRUD functions should be generalized for every Coredata model.